### ISSUE TITLE
shared reference version of `HashMap::raw_table`

### ISSUE THREAD

**Issue by digama0** [rust-lang/hashbrown#418]
I have a hashmap whose items require additional context to hash. As far as I know this can't be done with the normal Hash + Eq impls, so I switched from using `std::collections::HashMap` to `hashbrown::HashMap` to use the `raw_table` functions which allow open-coding the `get` and `insert` functions. Except, `get` can't be implemented without taking a `&mut Self` because `raw_table` requires `&mut` even though it has functions on it like `get` which don't require mutable access.

It is unfortunate that the name `raw_table` wasn't called `raw_table_mut`. Perhaps `raw_table_ref` is a possibility?

**Comment by Amanieu** [comment]
Yes, the naming is unfortunate, and this is a good candidate for a breaking change for hashbrown 0.14.

Have you considered using the `RawEntry` API on `HashMap`? Or just using `RawTable` directly?

**Comment by digama0** [comment]
Oh, I hadn't considered that you can construct `RawTable`s directly, that does solve my immediate issue. For `RawEntry`, I missed it because I'm actually using `HashSet`, not `HashMap`, and there is apparently no corresponding `raw_entry` function for `HashSet`. It is easy enough to switch to `HashMap<K, ()>` though.
